### DISCUSSION TITLE
use `set -o pipefail` by default

### DISCUSSION THREAD

**Discussion by meleu** [bashly-framework/bashly#595] [Ideas]
I noticed that by default bashly puts `set -e` in the `initialize` function. Which is great!

I suggest to also include `-o pipefail`, as it's a sane default.

NOTE: I'm aware about [the `strict` setting option](https://bashly.dannyb.co/usage/settings/#strict), what I'm suggesting here is to also add `-o pipefail` by default.

---

For reference: `set -o pipefail` makes the exit code of a sequence of piped commands to be the status of the last command with a non-zero status (or zero if all commands ran succesfully).

Here are two examples with and without `-o pipefail`.

`nopipefail.sh`:
```bash
#!/usr/bin/env bash

set -e

echo "Without '-o pipefail'"

erroneous_command() {
  cat /etc/passwd | invalid_command | echo "this echo's exit code is 0"

  echo "This line should not be printed"
}

erroneous_command "$@"
```

running it:
```
$ ./nopipefail.sh
Without '-o pipefail'
this echo's exit code is 0
./nopipefail.sh: line 8: invalid_command: command not found
This line should not be printed
```

As you can see :point_up_2: , the script continued the execution when it (normally) shouldn't.

`withpipefail.sh`:
```bash
#!/usr/bin/env bash

set -eo pipefail

echo "With '-o pipefail'"

erroneous_command() {
  cat /etc/passwd | invalid_command | echo "this echo's exit code is 0"

  echo "This line should not be printed"
}

erroneous_command "$@"
```

running it:
```
$ ./withpipefail.sh
With '-o pipefail'
./withpipefail.sh: line 8: invalid_command: command not found
this echo's exit code is 0
```

**Comment by DannyBen** [comment]
I have no way of knowing if this will negatively impact other users.

The rationale behind it being minimal, is that it should be the opposite of strict; the most minimal recommended / needed by bashly.

Fully customizable with sensible defaults.

**Reply by meleu** [reply]
fair enough.
let's close this one.
